### PR TITLE
server, session: add commitRetryLimit config.

### PR DESCRIPTION
**Pull request by disksing**
Some of our long-running tests want to increase the number of retries.
cc @coocood @shenli

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -218,10 +218,10 @@ func (s *session) doCommitWithRetry() error {
 	err := s.doCommit()
 	if err != nil {
 		if s.isRetryableError(err) {
-			// Transactions will retry 2 ~ 10 times.
+			// Transactions will retry 2 ~ commitRetryLimit times.
 			// We make larger transactions retry less times to prevent cluster resource outage.
 			txnSizeRate := float64(txnSize) / float64(kv.TxnTotalSizeLimit)
-			maxRetryCount := 10 - int(txnSizeRate*9.0)
+			maxRetryCount := commitRetryLimit - int(float64(commitRetryLimit)*txnSizeRate) + 1
```
Comment (zimulala) [inline session.go]: The max value of `maxRetryCount` is `maxRetryCount + 1` now.
I think we can remove `1`.

Comment (coocood) [inline session.go]: If removed, min retry would be 1.

Comment (shenli) [inline session.go]: https://github.com/pingcap/tidb/blob/master/session.go#L325
If maxRetryCount == 1, it will only retry once.

Comment (disksing) [inline session.go]: Old feature is kept as we've discussed. @zimulala @coocood

**Pull request merged**